python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -0,0 +1,257 @@
+        asset_graph = target_asset_partitions.asset_graph
+        return cls(
+            target_asset_partitions=target_asset_partitions,
+            roots_were_requested=False,
+            requested_asset_partitions=AssetGraphSubset(asset_graph),
+            materialized_asset_partitions=AssetGraphSubset(asset_graph),
+            failed_asset_partitions=AssetGraphSubset(asset_graph),
+            latest_storage_id=None,
+        )
+
+    @classmethod
+    def from_serialized(cls, serialized: str, asset_graph: AssetGraph) -> "AssetBackfillData":
+        storage_dict = json.loads(serialized)
+
+        return cls(
+            target_asset_partitions=AssetGraphSubset.from_storage_dict(
+                storage_dict["serialized_target_asset_partitions"], asset_graph
+            ),
+            roots_were_requested=storage_dict["roots_were_requested"],
+            requested_asset_partitions=AssetGraphSubset.from_storage_dict(
+                storage_dict["serialized_requested_asset_partitions"], asset_graph
+            ),
+            materialized_asset_partitions=AssetGraphSubset.from_storage_dict(
+                storage_dict["serialized_materialized_asset_partitions"], asset_graph
+            ),
+            failed_asset_partitions=AssetGraphSubset.from_storage_dict(
+                storage_dict["serialized_failed_asset_partitions"], asset_graph
+            ),
+            latest_storage_id=storage_dict["latest_storage_id"],
+        )
+
+    def serialize(self) -> str:
+        storage_dict = {
+            "roots_were_requested": self.roots_were_requested,
+            "serialized_target_asset_partitions": self.target_asset_partitions.to_storage_dict(),
+            "latest_storage_id": self.latest_storage_id,
+            "serialized_requested_asset_partitions": self.requested_asset_partitions.to_storage_dict(),
+            "serialized_materialized_asset_partitions": self.materialized_asset_partitions.to_storage_dict(),
+            "serialized_failed_asset_partitions": self.failed_asset_partitions.to_storage_dict(),
+        }
+        return json.dumps(storage_dict)
+
+
+def _get_implicit_job_name_for_assets(
+    asset_graph: ExternalAssetGraph, asset_keys: Sequence[AssetKey]
+) -> Optional[str]:
+    job_names = set(asset_graph.get_job_names(asset_keys[0]))
+    for asset_key in asset_keys[1:]:
+        job_names &= set(asset_graph.get_job_names(asset_key))
+
+    return next(job_name for job_name in job_names if is_base_asset_job_name(job_name))
+
+
+class AssetBackfillIterationResult(NamedTuple):
+    run_requests: Sequence[RunRequest]
+    backfill_data: AssetBackfillData
+
+
+def execute_asset_backfill_iteration_inner(
+    backfill_id: str,
+    asset_backfill_data: AssetBackfillData,
+    asset_graph: ExternalAssetGraph,
+    instance: DagsterInstance,
+) -> AssetBackfillIterationResult:
+    """Core logic of a backfill iteration. Has no side effects."""
```

Comment:
Extending this comment to describe the expected outcome of this function would be helpful

Reply:
👍